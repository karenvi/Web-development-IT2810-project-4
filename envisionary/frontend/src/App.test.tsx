import { render, screen } from '@testing-library/react';
import App from './App';
import { RecoilRoot } from 'recoil';
import { MockedProvider } from "@apollo/client/testing";
import { mocks } from './__test__/mocks/mocks';
import userEvent from '@testing-library/user-event';


it("App renders correctly", () => {
    const { container } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <RecoilRoot>
          <App/>
        </RecoilRoot>
      </MockedProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it('Tests the toggle from darkmode to lightmode', async () => {
    // ARRANGE
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <RecoilRoot>
          <App/>
        </RecoilRoot>
      </MockedProvider>
    )

    // ACT
    userEvent.click(screen.getByRole("checkbox", {name: "Theme switch"}))

    // ASSERT
    expect(await screen.findByRole("table")).toHaveStyle('backgroundColor: white; color: black')

  })


