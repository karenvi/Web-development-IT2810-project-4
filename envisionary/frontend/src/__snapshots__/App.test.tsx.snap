// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    class="App"
  >
    <header>
      <nav
        class="header"
        style="background-color: rgb(23, 42, 58);"
      >
        <div
          id="logoContainer"
        >
          <a
            href="#/"
            id="logoLink"
          >
            ENVISIONARY
          </a>
        </div>
        <div
          class="linkContainer"
        >
          <a
            class="link"
            href="#/"
          >
            Countries
          </a>
          <a
            class="link"
            href="#/info-page"
          >
            Information
          </a>
        </div>
      </nav>
    </header>
    <main
      class="MuiBox-root css-hbm2t4"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-1dwxtnu-MuiTypography-root"
        style="color: white;"
      >
        Search for a 
        country
      </h1>
      <div
        class="MuiBox-root css-1gtztm6"
        style="background-color: rgb(23, 42, 58); color: rgb(255, 255, 255);"
      >
        <form
          action="/"
          autocomplete="off"
          method="get"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-xveny1-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="header-search"
              id="header-search-label"
            >
              Search for country
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                data-testid="search-test"
                id="header-search"
                name="s"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Search for country
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </form>
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth css-1tp1e4i-MuiFormControl-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-154uufs-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            id="demo-simple-select-label"
          >
            Category:
          </label>
          <label
            for="demo-simple-select"
          >
            <span
              class="visually-hidden"
            >
              Select which category to search in
            </span>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-2h9eqm-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby="demo-simple-select-label demo-simple-select"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="demo-simple-select"
              role="button"
              tabindex="0"
            >
              Country
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              data-testid="select-test"
              tabindex="-1"
              value="Country"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Category:
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-1jzwcf2-MuiPaper-root-MuiTableContainer-root"
      >
        <table
          aria-label="Table of countries"
          class="MuiTable-root css-rqglhn-MuiTable-root"
          style="background-color: rgb(23, 42, 58); color: white;"
        >
          <thead
            class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-11sptls-MuiTableCell-root"
                colspan="2"
                scope="col"
              >
                <label
                  for="filter-category"
                >
                  <span
                    class="visually-hidden"
                  >
                    Sort by:
                  </span>
                </label>
                <div
                  class="MuiBox-root css-12zgw5k"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth css-jqb2fp-MuiFormControl-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="true"
                      id="select-filter-category"
                    >
                      Sort by:
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-2h9eqm-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                    >
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-labelledby="select-filter-category filter-category"
                        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                        id="filter-category"
                        role="button"
                        tabindex="0"
                      >
                        Ascending country
                      </div>
                      <input
                        aria-hidden="true"
                        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                        tabindex="-1"
                        value="Country-asc"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                        data-testid="ArrowDropDownIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M7 10l5 5 5-5z"
                        />
                      </svg>
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-14lo706"
                        >
                          <span>
                            Sort by:
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                colspan="2"
                scope="col"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                Hide unreviewed countries
                <span
                  class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-root MuiCheckbox-colorPrimary css-71ytt9-MuiButtonBase-root-MuiCheckbox-root"
                >
                  <input
                    aria-label="controlled"
                    class="PrivateSwitchBase-input css-1m9pwf3"
                    data-indeterminate="false"
                    type="checkbox"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="CheckBoxOutlineBlankIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </span>
              </th>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-11sptls-MuiTableCell-root"
                scope="col"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                Country
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                scope="col"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                Continent
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                scope="col"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                Population (2022)
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1o98s2t-MuiTableCell-root"
                scope="col"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                Area (km²)
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-34nofg-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <th
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium pointer css-1ex1afd-MuiTableCell-root"
                scope="row"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </th>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeMedium pointer css-177gid-MuiTableCell-root"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <span
                  class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1n7gpw1-MuiSkeleton-root"
                />
                 
              </td>
            </tr>
            <tr
              class="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
              style="background-color: rgb(23, 42, 58); color: white;"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
                colspan="4"
                style="background-color: rgb(23, 42, 58); color: white;"
              >
                <div
                  class="MuiGrid-root MuiGrid-container css-zo8441-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root css-gfmvt9-MuiGrid-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-142aol9-MuiButtonBase-root-MuiButton-root"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      Previous
                    </button>
                  </div>
                  <div
                    class="MuiGrid-root css-1yi6br4-MuiGrid-root"
                  >
                    Page 
                    1
                  </div>
                  <div
                    class="MuiGrid-root css-1ivdjg1-MuiGrid-root"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-142aol9-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Next
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </main>
    <div
      class="MuiBox-root css-1ovvwn9"
    >
      <div
        class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        >
          <span
            class="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
          >
            <span
              class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-5ryogn-MuiButtonBase-root-MuiSwitch-switchBase"
            >
              <input
                aria-label="Theme switch"
                checked=""
                class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                type="checkbox"
              />
              <span
                class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </span>
            <span
              class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            Switch to light theme?
          </span>
        </label>
      </div>
    </div>
  </div>
</div>
`;
